packages/opentelemetry-tracing/src/utility.ts

```diff
@@ -21,14 +21,34 @@ import {
   DEFAULT_MAX_LINKS_PER_SPAN,
 } from './config';
 import { TracerConfig } from './types';
+import {
+  ParentOrElseSampler,
+  ProbabilitySampler,
+  getEnv,
+} from '@opentelemetry/core';
 
 /**
  * Function to merge Default configuration (as specified in './config') with
  * user provided configurations.
  */
 export function mergeConfig(userConfig: TracerConfig) {
   const traceParams = userConfig.traceParams;
-  const target = Object.assign({}, DEFAULT_CONFIG, userConfig);
+  const otelSamplingProbability = getEnv().OTEL_SAMPLING_PROBABILITY;
+  const validOtelSamplingProbability =
+    otelSamplingProbability &&
+    otelSamplingProbability >= 0 &&
```

Comment:
shoot, thanks.

Reply:
actually no, its a number
![image](https://user-images.githubusercontent.com/24660299/87842096-d77a4780-c877-11ea-9a74-ddd43efa28b5.png)

Reply:
but the logic pasted is slightly flawed, looking into it

Reply:
I think the first condition is not needed as second `otelSamplingProbability >=0` is already enough

Reply:
besides that this is already checked in env so in fact OTEL_SAMPLING_PROBABILITY will always be a number between 0 and 1 (default is set to 1)

Reply:
Thanks @obecny that's a good point! I will only check if it is less than one, and if not, will use the AlwaysOnSampler() as per spec's recommendation

```js
    otelSamplingProbability !== undefined && otelSamplingProbability < 1
```
is the check since according to ts it could be undefined.

Reply:
after this PR -> https://github.com/open-telemetry/opentelemetry-js/pull/1328  ts will not complain about undefined